src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/SemanticTokens.cs

```diff
@@ -49,13 +50,48 @@ internal partial class RazorCustomMessageTarget
             return new ProvideSemanticTokensResponse(tokens: null, hostDocumentSyncVersion: csharpDoc.HostDocumentSyncVersion ?? -1);
         }
 
-        using var _ = ListPool<Task<ReinvocationResponse<VSSemanticTokensResponse>?>>.GetPooledObject(out var requestTasks);
-
         semanticTokensParams.TextDocument.Uri = csharpDoc.Uri;
         var textBuffer = csharpDoc.Snapshot.TextBuffer;
 
         cancellationToken.ThrowIfCancellationRequested();
+        ReinvocationResponse<VSSemanticTokensResponse?>? result = null;
+        if (semanticTokensParams.UsePreciseRanges)
+        {
+            var languageServerName = RazorLSPConstants.RazorCSharpLanguageServerName;
```

Comment:
imo I'd break the two branches here to their own methods, it will make it a little easier to read